src/components/ChannelSearch/SearchResults.tsx

```diff
@@ -1,54 +1,153 @@
 import React, { useCallback, useEffect, useState } from 'react';
 
+import { ChannelOrUserResponse, isChannel } from './utils';
+
 import { Avatar } from '../Avatar/Avatar';
 import { useBreakpoint } from '../Message/hooks/useBreakpoint';
 
 import { useTranslationContext } from '../../context/TranslationContext';
 
-import type { UserResponse } from 'stream-chat';
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+} from '../../types/types';
+
+export type DropdownContainerProps<
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends DefaultCommandType = DefaultCommandType,
+  Ev extends DefaultEventType = DefaultEventType,
+  Me extends DefaultMessageType = DefaultMessageType,
+  Re extends DefaultReactionType = DefaultReactionType,
+  Us extends DefaultUserType<Us> = DefaultUserType
+> = {
+  results: ChannelOrUserResponse<At, Ch, Co, Ev, Me, Re, Us>[];
+  SearchResultItem: React.ComponentType<SearchResultItemProps<At, Ch, Co, Ev, Me, Re, Us>>;
+  selectResult: (result: ChannelOrUserResponse<At, Ch, Co, Ev, Me, Re, Us>) => Promise<void> | void;
+  focusedUser?: number;
+};
+
+const DefaultDropdownContainer = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends DefaultChannelType = DefaultChannelType,
+  Co extends DefaultCommandType = DefaultCommandType,
+  Ev extends DefaultEventType = DefaultEventType,
+  Me extends DefaultMessageType = DefaultMessageType,
+  Re extends DefaultReactionType = DefaultReactionType,
+  Us extends DefaultUserType<Us> = DefaultUserType
+>(
+  props: DropdownContainerProps<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const { focusedUser, results, SearchResultItem = DefaultSearchResultItem, selectResult } = props;
 
-import type { DefaultUserType } from '../../types/types';
+  return (
+    <div>
+      {results.map((result: ChannelOrUserResponse<At, Ch, Co, Ev, Me, Re, Us>, index: number) => (
```

Comment:
`result` type should be inferred from `results` so don't need to include, same with `index`